app/src/main/java/com/randomappsinc/studentpicker/importdata/FileParsingManager.java

```diff
@@ -0,0 +1,172 @@
+                    }
+                    if (reader != null) {
+                        reader.close();
+                    }
+                } catch (IOException ignored) {}
+            }
+
+            listener.onFileParsingSuccess(listNameText, namesText.toString());
+        });
+    }
+
+    void extractNameListFromCsv(Context context, Uri fileUri) {
+        backgroundHandler.post(() -> {
+            Cursor cursor = context.getContentResolver().query(
+                    fileUri,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null);
+
+            String listNameText = "";
+            if (cursor != null && cursor.moveToFirst()) {
+                String displayName = cursor.getString(cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME));
+                listNameText = displayName.replace(".csv", "");
+                cursor.close();
+            }
+
+            StringBuilder namesText = new StringBuilder();
+            InputStream inputStream = null;
+            BufferedReader reader = null;
+            try {
+                inputStream = context.getContentResolver().openInputStream(fileUri);
+                if (inputStream == null) {
+                    throw new IOException("Unable to find .csv file!");
+                }
+                reader = new BufferedReader(new InputStreamReader(inputStream));
+                CsvReader csvReader = new CsvReader();
+                CsvParser csvParser = csvReader.parse(reader);
+
+                CsvRow row;
+                while ((row = csvParser.nextRow()) != null) {
+                    String firstName = "";
+                    try {
+                        firstName = row.getField(0);
+                    } catch (IndexOutOfBoundsException ignored) {}
+                    String lastName = "";
+                    try {
+                        lastName = row.getField(1);
+                    } catch (IndexOutOfBoundsException ignored) {}
+
+                    if (namesText.length() > 0) {
+                        namesText.append("\n");
+                    }
+                    namesText.append(firstName);
+                    namesText.append(" ");
+                    namesText.append(lastName);
+                }
+            } catch (IOException | ArrayIndexOutOfBoundsException exception) {
+                listener.onFileParsingFailure(R.string.csv_read_failed);
+                try {
+                    if (inputStream != null) {
+                        inputStream.close();
+                    }
+                    if (reader != null) {
+                        reader.close();
+                    }
+                } catch (IOException ignored) {}
+            }
+
+            listener.onFileParsingSuccess(listNameText, namesText.toString());
+        });
+    }
+
+    void saveNameList(Context context, String newListName, String[] allNames) {
```

Comment:
With this method, FileParsingManager as a name doesn't capture the scope of this class anymore. I think we should rename this to FileImportManager.